engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -168,8 +171,10 @@ class WorkflowManagerActor(backend: Backend, config: Config)
 
   when (Running) {
     case Event(SubmitWorkflow(source), _) =>
-      val updatedData = submitWorkflow(source, replyTo = Option(sender), id = None)
-      stay() using updatedData
+      submitWorkflow(source, replyTo = Option(sender), id = None) onSuccess {
+        case updatedData: WorkflowManagerData => self ! UpdateWorkflowManagerState(updatedData)
+      }
+      stay()
```

Comment:
If the message is more event-sourcish in meaning and communicated "add A to whatever the state is at the time you receive this", this should work.